### DISCUSSION TITLE
[HELP Request] Identify if there is a Note Toolbar in the active Note

### DISCUSSION THREAD

**Discussion by FelipeRearden** [chrisgurney/obsidian-note-toolbar#23] [Q&A] [answered]
Hello @chrisgurney !!!

I need some help about a templater script that I am trying to build…

In a nutshell: my script will execute different commands based on the fact that the active note has a **Note Toolbar** or not. 

Using the idea from this post below I created this templater script:
https://github.com/pjeby/pane-relief/issues/45#issuecomment-1507844844

```
<%*
if (activeDocument.body.find(".cg-note-toolbar-container")) {
new Notice('ACTIVE', 10000);
} else {
new Notice('NO NOTE TOOLBAR', 10000);
}
%>
```
BUT even if there is NO NOTE TOOLBAR in the active note (YAML or FOLDER) the script always return `true` .

<hr>

**MY QUESTION** would mind sharing some element that could help to add to this script that is not present when the active note dont have a NOTE TOOLBAR?

I am 100% open to a new approach if you want :) As you know my code skills are bad.

<hr>

Desired workflow: I want to run the command `app.commands.executeCommandById('note-toolbar:focus') ` from a hotkey only when there is a Note Toolbar is the active note. If not it will run another command. 

something like this…

```

<%*
if (activeDocument.body.find(".cg-note-toolbar-container")) {
new Notice('ACTIVE', 10000);
app.commands.executeCommandById('note-toolbar:focus')
} else {
new Notice('NO NOTE TOOLBAR', 10000);
}
%>
```

Thanks for the help 🙏

**Comment by chrisgurney** [comment, marked as answer]
@FelipeRearden You were close! I believe what you're looking for (if I understand your use case) should be this:

`.workspace-leaf.mod-active .cg-note-toolbar-container`

Let me know if this works!

**Reply by FelipeRearden** [reply]
AMAZING !!!! it worked perfect !!!

How I could forgot about `.mod-active` 🤦🏻‍♂️

Thank you very much Chris for the help 💙

**Comment by chrisgurney** [comment]
@FelipeRearden I'm really curious about your use case!

Why do you want to immediately focus the keyboard in the toolbar? What item in the toolbar would you then use next? Would this be only in Reading mode?

**Reply by FelipeRearden** [reply]
> Would this be only in Reading mode?

No, this templater is will be my command to Focus on Note Toolbar in LP and RM. The main idea behind this templater is to save a hotkey for specific notes that will not have a Note Toolbar

> Why do you want to immediately focus the keyboard in the toolbar? What item in the toolbar would you then use next?

Great question. I will only use Note Toolbar from the Keyboard. This way I need to immediately focus to use Note Toolbar. 

<hr> 

Ask more questions if you need, I am all ears !!!!

**Reply by chrisgurney** [reply]
> The main idea behind this templater is to save a hotkey for specific notes that will not have a Note Toolbar

What does the hotkey do for those notes (that don't have the toolbar)?

> I will only use Note Toolbar from the Keyboard. This way I need to immediately focus to use Note Toolbar.

Got it! I'm actually now considering an enhancement that runs any number of commands, that you define, on toolbar load.

As an example, I just committed three new commands to hide, show, and toggle the Properties section (will release in a day or two). So I'm envisioning that if you wanted to hide the Properties for given notes, you'd just setup the toolbar to do it for you, and you can add a toggle button (which works great) to open/close the Properties to your toolbar. This same feature would let you run the Focus command, in your case.

I'm curious what commands you have in your toolbar!

**Reply by chrisgurney** [reply]
@FelipeRearden Update here:

> I'm actually now considering an enhancement that runs any number of commands, that you define, on toolbar load.

Unfortunately I'm running into issues getting commands to fire predictably, based on file open, partly due to how Reading/Preview mode is rendered (compared to Source view), and caching, etc.

I think users might ultimately want preferences around command execution as well, based on the command being run. 

All that said, I suspect such an enhancement might result in issues being logged about unpredictability.

So I'm dropping this for now. Maybe in the future there might be another way to execute a script or commands in a predictable way.

**Reply by FelipeRearden** [reply]
> What does the hotkey do for those notes (that don't have the toolbar)?

Since it is a few notes and very specific I am gonna run a different command for each Note. The templater script will support this management.

> I'm curious what commands you have in your toolbar!

This is a big WIP that I am building step by step. I have to decide wihich commands will be transferred from hotkeys to Toolbars 

In a nutshell, my vault is based on Folders and the Workspaces core plugin. This way I am gonna have at least 8 Toolbars, one for each workspace at least.

The workspace commands are the crazy ideas that I have to manage and switch text between notes.

In addition, I am gonna have a standard Toolbar which has Search `items`:
- Search in File;
- Search in Vault;
- Search in my MacOS/iPadOS (dont know if is feasible yet);
- Search in Wikipedia;
- Dictionary + Translate;

This way, my Toolbar will be STANDARD + WORKSPACE items.

 This is the  basic idea so far. I am so happy to be able to have Note Toolbar instead of a command  Modal.

> Got it! I'm actually now considering an enhancement that runs any number of commands, that you define, on toolbar load.
> …
>
> So I'm dropping this for now. Maybe in the future there might be another way to execute a script or commands in a predictable way.

I think it is a great idea but seems to be a big challenge and seems to be a great idea to drop this for now. We are gonna have more ideas to could be easier to implement :)

<hr> 

I wish you a fantastic start of week and a great day ☀️
